lib/active_merchant/billing/gateways/payflow_express.rb

```diff
@@ -69,6 +69,11 @@ class PayflowExpressGateway < Gateway
       self.homepage_url = 'https://www.paypal.com/cgi-bin/webscr?cmd=xpt/merchant/ExpressCheckoutIntro-outside'
       self.display_name = 'PayPal Express Checkout'
 
+      def redirect_url_for(token, options = {})
+        cmd = options[:mobile] ? '_express-checkout-mobile' : '_express-checkout'
+        "#{base_redirect_url_for(token, options)}&cmd=#{cmd}"
```

Comment:
Instead of doing this, why not have `base_redirect_url_for` accept a `cmd` argument (that could be nil)?